File: tiq-e/others/6/main.py
from time import perf_counter
from typing import List


class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        result = (n * (n + 1)) // 2
        for num in nums:
            result -= num
        return result


x = [3, 0, 1]

start = perf_counter()
result = Solution().missingNumber(x)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))

File: tiq-e/others/6/main.rs
use std::time::Instant;

struct Solution;

impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let n = nums.len() as i32;
        let sum: i32 = nums.iter().sum();
        n * (n + 1) / 2 - sum
    }
}

fn main() {
    let x = vec![3, 0, 1];

    let start = Instant::now();
    let result = Solution::missing_number(x);
    let end = start.elapsed().as_micros();
    println!("Duration: {:?}", end);
    println!("Result: {:?}", result);
}
